### ISSUE TITLE
Process Tomography: RuntimeWarning: invalid value encountered in true_divide   freqs = counts / shots

### ISSUE THREAD

**Issue by MarcelloCa** [Qiskit/qiskit-terra#1539]
### Informations

- **Qiskit Terra version**: 0.6.1
- **Python version**: 3.7
- **Operating system**: osx

### What is the current behavior?

Since yesterday evening, I'm having the following error by performing a process tomography for the teleportation process as here [https://github.com/Qiskit/qiskit-terra/issues/1432]
```
/Applications/anaconda3/lib/python3.7/site-packages/qiskit/tools/qcvv/tomography.py:806: RuntimeWarning: invalid value encountered in true_divide
  freqs = counts / shots
```

The Traceback is as follows:
```
---------------------------------------------------------------------------
LinAlgError                               Traceback (most recent call last)
<ipython-input-2-9a29d3ef1c00> in <module>()
     17     tomo_result = tomo_job.result()
     18     process_data = tomo.tomography_data(tomo_result, qc.name, tomo_set)
---> 19     exp_choi = tomo.fit_tomography_data(process_data, options={'trace':2})
     20     #plot_state(id_choi)
     21     #plot_state(exp_choi)

/Applications/anaconda3/lib/python3.7/site-packages/qiskit/tools/qcvv/tomography.py in fit_tomography_data(tomo_data, method, options)
    742             # Use wizard method to constrain positivity
    743             epsilon = __get_option('epsilon', options)
--> 744             rho = __wizard(rho, epsilon=epsilon)
    745         return rho
    746     else:

/Applications/anaconda3/lib/python3.7/site-packages/qiskit/tools/qcvv/tomography.py in __wizard(rho, epsilon)
    895     dim = len(rho)
    896     rho_wizard = np.zeros([dim, dim])
--> 897     v, w = np.linalg.eigh(rho)  # v eigenvecrors v[0] < v[1] <...
    898     for j in range(dim):
    899         if v[j] < epsilon:

/Applications/anaconda3/lib/python3.7/site-packages/numpy/linalg/linalg.py in eigh(a, UPLO)
   1408 
   1409     signature = 'D->dD' if isComplexType(t) else 'd->dd'
-> 1410     w, vt = gufunc(a, signature=signature, extobj=extobj)
   1411     w = w.astype(_realType(result_t), copy=False)
   1412     vt = vt.astype(result_t, copy=False)

/Applications/anaconda3/lib/python3.7/site-packages/numpy/linalg/linalg.py in _raise_linalgerror_eigenvalues_nonconvergence(err, flag)
     93 
     94 def _raise_linalgerror_eigenvalues_nonconvergence(err, flag):
---> 95     raise LinAlgError("Eigenvalues did not converge")
     96 
     97 def _raise_linalgerror_svd_nonconvergence(err, flag):

LinAlgError: Eigenvalues did not converge
```

Before that, I ran over one hundred process tomographies in the last couple of weeks without any error.

### Steps to reproduce the problem

```
tomo_result = tomo_job.result()
process_data = tomo.tomography_data(tomo_result, qc.name, tomo_set)
exp_choi = tomo.fit_tomography_data(process_data, options={'trace':2})
```

**Comment by MarcelloCa** [comment]
It seems like there has not been any shoot at all.
Here the code to reproduce the error (which appears also if measured qubits and prepared qubits are the same.

```
import numpy as np
import time

# importing Qiskit
from qiskit import Aer, IBMQ
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
from qiskit import register, execute

# import backend utilities
from qiskit.backends import JobStatus

# import tomography library
import qiskit.tools.qcvv.tomography as tomo

# useful additional packages 
from qiskit.tools.visualization import plot_state
from qiskit.tools.qi.qi import *
from qiskit.wrapper.jupyter import *
#from qiskit.backends.ibmq import least_busy

# Load saved IBMQ accounts
IBMQ.load_accounts()

# Creating circuit for tomography
q = QuantumRegister(1)
c = ClassicalRegister(1)
qc = QuantumCircuit(q, c, name='teleportation')

# inizializing qubit ⓪ to be teleported
qc.u1(0, q[0])
qc.barrier(q)

# Construct state tomography set for measurement of qubits [0, 1] in the Pauli basis
tomo_set = tomo.process_tomography_set([0])
# Add the state tomography measurement circuits to the Quantum Program
tomo_circuits = tomo.create_tomography_circuits(qc, q, c, tomo_set)

backend = IBMQ.get_backend('ibmq_16_melbourne') # the device to run on
shots = 1024    # the number of shots in the experiment
max_credits = 8

qubit=0
params = backend.properties()['qubits'][qubit]
pulse_length=params['gateTime']['value'] # single-qubit gate time 
buffer_length=params['buffer']['value'] # spacing between pulses
unit = params['gateTime']['unit']

print('pulse_length: ', pulse_length, unit)
print('buffer_length: ', buffer_length, unit)

# Run the simulation
tomo_job = execute(tomo_circuits, backend=backend, shots=shots, max_credits=max_credits)

lapse = 0
interval = 10
while tomo_job.status().name != 'DONE':
    print('Status @ {} seconds'.format(interval * lapse))
    print('\t', tomo_job.status())
    print('\t', tomo_job.queue_position())
    time.sleep(interval)
    lapse += 1

tomo_result = tomo_job.result()
print(tomo_result)

process_data = tomo.tomography_data(tomo_result, qc.name, tomo_set)
print(process_data)

#choi_fit = tomo.fit_tomography_data(process_data,method='leastsq', options={'trace':2})
choi_fit = tomo.fit_tomography_data(process_data, options={'trace':2})
```

and this is the output of 'print(tomo_result)'

```
{'data': [{'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('X', 0)], 'prep': [('S', 0)]}, '1': {'meas': [('X', 1)], 'prep': [('S', 0)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Y', 0)], 'prep': [('S', 0)]}, '1': {'meas': [('Y', 1)], 'prep': [('S', 0)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Z', 0)], 'prep': [('S', 0)]}, '1': {'meas': [('Z', 1)], 'prep': [('S', 0)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('X', 0)], 'prep': [('S', 1)]}, '1': {'meas': [('X', 1)], 'prep': [('S', 1)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Y', 0)], 'prep': [('S', 1)]}, '1': {'meas': [('Y', 1)], 'prep': [('S', 1)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Z', 0)], 'prep': [('S', 1)]}, '1': {'meas': [('Z', 1)], 'prep': [('S', 1)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('X', 0)], 'prep': [('S', 2)]}, '1': {'meas': [('X', 1)], 'prep': [('S', 2)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Y', 0)], 'prep': [('S', 2)]}, '1': {'meas': [('Y', 1)], 'prep': [('S', 2)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Z', 0)], 'prep': [('S', 2)]}, '1': {'meas': [('Z', 1)], 'prep': [('S', 2)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('X', 0)], 'prep': [('S', 3)]}, '1': {'meas': [('X', 1)], 'prep': [('S', 3)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Y', 0)], 'prep': [('S', 3)]}, '1': {'meas': [('Y', 1)], 'prep': [('S', 3)]}}}, {'counts': {'0': 0, '1': 0}, 'shots': 0, 'circuit': {'0': {'meas': [('Z', 0)], 'prep': [('S', 3)]}, '1': {'meas': [('Z', 1)], 'prep': [('S', 3)]}}}], 'meas_basis': {'X': [array([[0.5, 0.5],
       [0.5, 0.5]]), array([[ 0.5, -0.5],
       [-0.5,  0.5]])], 'Y': [array([[ 0.5+0.j , -0. -0.5j],
       [ 0. +0.5j,  0.5+0.j ]]), array([[ 0.5+0.j ,  0. +0.5j],
       [-0. -0.5j,  0.5+0.j ]])], 'Z': [array([[1, 0],
       [0, 0]]), array([[0, 0],
       [0, 1]])]}, 'prep_basis': {'S': [array([[1, 0],
       [0, 0]]), array([[0.33333333, 0.47140452],
       [0.47140452, 0.66666667]]), array([[ 0.33333333+0.j        , -0.23570226+0.40824829j],
       [-0.23570226-0.40824829j,  0.66666667+0.j        ]]), array([[ 0.33333333+0.j        , -0.23570226-0.40824829j],
       [-0.23570226+0.40824829j,  0.66666667+0.j        ]])]}}
```

**Comment by nonhermitian** [comment]
I cannot reproduce your error.  The above code ran fine.  Note however that I have run in on `qiskit-terra 0.7` that is being released soon.

**Comment by MarcelloCa** [comment]
I tried once more and it still doesn't work (counts and shots equal to 0 within the tomo_job.result() )

I used qiskit 0.6.1 with tomography.py patched as from @chriseclectic fix [here](https://github.com/Qiskit/qiskit-terra/issues/1432).

Updated: same issue with the original tomography.py

**Comment by MarcelloCa** [comment]
I'm not sure eigensolver is the issue.

Look at the awkward tomo_result:
- zero shots for each measurement
- hence, with zero counts for each of the two possible measurement outputs

There maybe something wrong within the result

**Comment by MarcelloCa** [comment]
problem solved in 0.7.0..

FYI, the issue in 0.6.1 was likely due to "get_counts" produced with hex value since monday
